src/Nancy/Owin/NancyMiddleware.cs

```diff
@@ -70,11 +70,11 @@ public static MidFunc UseNancy(NancyOptions options = null)
                         var owinRequestHost = GetHeader(owinRequestHeaders, "Host") ?? Dns.GetHostName();
                         var owinUser = GetUser(environment); 
 
-                        byte[] certificate = null;
+                        X509Certificate2 certificate = null;
                         if (options.EnableClientCertificates)
                         {
-                            var clientCertificate = Get<X509Certificate>(environment, "ssl.ClientCertificate");
-                            certificate = (clientCertificate == null) ? null : clientCertificate.GetRawCertData();
+                            var clientCertificate = Get<X509Certificate2>(environment, "ssl.ClientCertificate");

```

Comment:
Assume so

On Sunday, 21 February 2016, Andreas Håkansson <notifications@github.com>
wrote:

> In src/Nancy/Owin/NancyMiddleware.cs
> <https://github.com/NancyFx/Nancy/pull/2293#discussion_r53573887>:
>
> >                          if (options.EnableClientCertificates)
> >                          {
> > -                            var clientCertificate = Get<X509Certificate>(environment, "ssl.ClientCertificate");
> > -                            certificate = (clientCertificate == null) ? null : clientCertificate.GetRawCertData();
> > +                            var clientCertificate = Get<X509Certificate2>(environment, "ssl.ClientCertificate");
>
> This is guaranteed to work because X509Certificate2 inherits from
> X509Certificate ?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/NancyFx/Nancy/pull/2293/files#r53573887>.
>